osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -16,17 +16,11 @@ namespace osu.Game.Rulesets
     /// </summary>
     public class RulesetStore : DatabaseBackedStore
     {
-        private static readonly Dictionary<Assembly, Type> loaded_assemblies = new Dictionary<Assembly, Type>();
+        private static readonly Lazy<Dictionary<Assembly, Type>> loaded_assemblies = new Lazy<Dictionary<Assembly, Type>>(() => loadRulesets());
```

Comment:
Can you guarantee that accesses to this are thread-safe? (as in, this can't be called multiple times and cause races) If not, might be worth sticking a `isThreadSafe: false` on this.

Reply:
The initialization logic is thread safe. The only thing I could imagine creating problems is Assembly.Load. But that method is also thread safe according to [here](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly?view=netframework-4.8#thread-safety)

So this should be fine.